File: common/src/main/scala/com/taskadapter/connector/common/FieldPrettyNameBuilder.scala
package com.taskadapter.connector.common

import com.taskadapter.model.Field

object FieldPrettyNameBuilder {
  def getPrettyFieldName(f: Field[_]): String = {
    if (f.getClass.getSimpleName.startsWith("Custom")) {
      f.toString
    } else {
      f.getClass.getSimpleName.replace("$", "")
    }
  }
}

File: common/src/main/java/com/taskadapter/connector/common/FieldPrettyNameBuilder.java
package com.taskadapter.connector.common;

import com.taskadapter.model.Field;

public class FieldPrettyNameBuilder {
    public static String getPrettyFieldName(Field<?> f) {
        if (f.getClass().getSimpleName().startsWith("Custom")) {
            return f.toString();
        }
        return f.getClass().getSimpleName().replace("$", "");
    }
}
